torch/optim/rprop.py

```diff
@@ -110,7 +110,12 @@ def step(self, closure=None):
                 if len(state) == 0:
                     state['step'] = 0
                     state['prev'] = torch.zeros_like(p, memory_format=torch.preserve_format)
-                    state['step_size'] = grad.new().resize_as_(grad).fill_(group['lr'])
+                    if p.dtype.is_complex:
+                        # Complex Number should be as if they are two independent real numbers.
+                        # Hence the step_size shouldn't be zero for imaginary part.
+                        state['step_size'] = grad.new().resize_as_(grad).fill_(complex(group['lr'], group['lr']))
```

Comment:
nit: why not keep the step_size real and just make it twice as big by using `.resize_as_(torch.view_as_real(grad))...`

Reply:
Down the line there are inplace operations with step_size
https://github.com/pytorch/pytorch/blob/eff28d61c9c961e6c2724b78f57441ee2e3e40cb/torch/optim/rprop.py#L197-L198

Here sign is complex tensor viewed as real and `step_size` could turn out to be different for real and imaginary part after a few steps.

Reply:
Sure, let's ignore that.